bin/subst.ml

```diff
@@ -110,15 +110,18 @@ let subst_string s path ~map =
 ;;
 
 let subst_file path ~map =
-  let s = Io.read_file path in
-  let s =
-    if Path.is_root (Path.parent_exn path) && Package.is_opam_file path
-    then "version: \"%%" ^ "VERSION_NUM" ^ "%%\"\n" ^ s
-    else s
-  in
-  match subst_string s ~map path with
-  | None -> ()
-  | Some s -> Io.write_file path s
+  match Io.read_file path with
+  | exception Io.Large_file ->
+    User_warning.emit [ Pp.textf "Ignoring large file: %s" (Path.to_string path) ]
```

Comment:
The warning is very useful, that way we can also monitor via bug reports if this is a problem that we would need to solve in a different way, e.g. by streaming the file.

I think it would be useful to add a `hint` to tell the user that the file is too large for 32 bit platforms.

Reply:
done. not too sure about the wording, let me know what you think

Reply:
Looks good to me!